cardano-db/src/Cardano/Db/Migration/Extra/CosnumedTxOut/Queries.hs

```diff
@@ -202,40 +206,67 @@ countConsumed = do
     pure countRows
   pure $ maybe 0 unValue (listToMaybe res)
 
+deleteAndUpdateConsumedTxOut :: MonadIO m => Trace IO Text -> Word64 -> ReaderT SqlBackend m ()
+deleteAndUpdateConsumedTxOut trce blockNoDiff = do
+  maxTxInId <- findMaxTxInId blockNoDiff
+  case maxTxInId of
+    Left errMsg -> liftIO $ logInfo trce $ "No tx_out was deleted: " <> errMsg
+    Right mxtxid -> do
+      migrateNextPage mxtxid 0
+  where
+    migrateNextPage :: MonadIO m => TxInId -> Word64 -> ReaderT SqlBackend m ()
+    migrateNextPage mxTxInId offst = do
+      page <- getInputPage offst pageSize
+      -- before page txinId delete after create extra collumn and update
+      mapM_ (handlePage mxTxInId) page
+      when (fromIntegral (length page) == pageSize) $
+        migrateNextPage mxTxInId $!
+          offst
+            + pageSize
+
+handlePage :: MonadIO m => TxInId -> (TxInId, TxId, Word64) -> ReaderT SqlBackend m ()
+handlePage maxTxInId (txInId, txId, index)
+  | txInId <= maxTxInId = deleteTxOutConsumed txId index
+  | otherwise = do
+      createConsumedTxOut
```

Comment:
I think this can be called more than once, ie for consecutive pages after `maxTxInId`, possibly causing issues.

Reply:
ah ok, we have access to `wPruneTxOutPreviouslySet` so can pass that through as a check?

Reply:
I don't think `wPruneTxOutPreviouslySet` can help here. It must be False, otherwise this code is not executed at all. 
`handlePage` being an iterative function can call `createConsumedTxOut` many times. Probably a new local flag can help.

Reply:
sorry I edited it, we have `hasConsumedField` which should do the trick